javav2/example_code/opensearch/src/main/java/com/example/search/HelloOpenSearch.java

```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.example.search;
+
+// snippet-start:[opensearch.java2.hello.main]
+import software.amazon.awssdk.services.opensearch.OpenSearchAsyncClient;
+import software.amazon.awssdk.services.opensearch.model.ListVersionsRequest;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Before running this Java V2 code example, set up your development
+ * environment, including your credentials.
+ *
+ * For more information, see the following documentation topic:
+ *
+ * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
+ */
+public class HelloOpenSearch {
+    public static void main(String[] args) {
+        try {
+            CompletableFuture<Void> future = listVersionsAsync();
+            future.join();
+            System.out.println("Versions listed successfully.");
+        } catch (RuntimeException e) {
+            System.err.println("Error occurred while listing versions: " + e.getMessage());
+        }
+    }
+
+    private static OpenSearchAsyncClient getAsyncClient() {
+        return OpenSearchAsyncClient.builder().build();
+    }
+
+    public static CompletableFuture<Void> listVersionsAsync() {
+        ListVersionsRequest request = ListVersionsRequest.builder()
+            .maxResults(10)
+            .build();
+
+        return getAsyncClient().listVersions(request).thenAccept(response -> {
```

Comment:
This is not using the paginated api.

Reply:
This does not work well with Async client